torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -82,6 +87,18 @@ LTCTensorImpl::LTCTensorImpl(LazyTensor&& tensor)
   // This is a temporary fix for a PyTorch core issue,
   // according to https://github.com/pytorch/xla/pull/2682.
   is_non_overlapping_and_dense_ = false;
+
+  auto rank = tensor_->shape().Get().sizes().size();
+  std::vector<int64_t> sym_ints;
+  sym_ints.reserve(rank);
+
+  for (auto i: c10::irange(rank)) {
+    auto dim_node = MakeNode<SizeNode>(this->tensor_->GetIrValue(), i);
+    auto sn = std::make_shared<torch::lazy::SymbolicIntNode>(dim_node);
+    sym_ints.push_back(sn->toSymInt().data());
+  }
+
+  sizes_and_strides_.set_sizes(sym_ints);
```

Comment:
Hmm, so `sym_int`  can be used to look up `SymbolicIntNode`(per dimension), and `syn_int` lives within `TensorImpl` instead of `LazyTensor` class?

Reply:
> Hmm, so sym_int can be used to look up SymbolicIntNode

> syn_int lives within TensorImpl

yes, symint would live in TensorImpl. That's the gist of @ezyang proposal. We will be making SymInts a first class citizen.